blogapp/views.py

```diff
@@ -28,20 +29,24 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
     def retrieve(self, request, pk=None, *args, **kwargs):
-        queryset = BlogPost.objects.all()
-        blogpost = get_object_or_404(queryset, pk=pk)
+        blogpost = get_object_or_404(BlogPost, pk=pk)
         serializer = BlogPostSerializer(blogpost)
         return Response(serializer.data)
 
     def update(self, request, pk=None, *args, **kwargs):
-        blogpost = BlogPost.objects.get(pk=pk)
-        serializer = BlogPostSerializer(blogpost, data=request.data)
-        if serializer.is_valid():
+        try:
+            blogpost = BlogPost.objects.get(pk=pk)
+            serializer = BlogPostSerializer(blogpost, data=request.data)
+            serializer.is_valid(raise_exception=True)
             serializer.save()
             return Response(serializer.data)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        except BlogPost.DoesNotExist:
+            return Response({"detail": "Blogpost not found."}, status=status.HTTP_404_NOT_FOUND)
```

Comment:
the fact that you continue to use try and catch syuggest you don't understand what `self.get_object` or `get_object_or_404` is doing